lightning/src/ln/channelmanager.rs

```diff
@@ -12531,72 +12326,19 @@ where
 					},
```

Comment:
10 lines up -- Can we move the payment hash/secret creation into `create_response_for_invoice_request`? Or does Fedimint need to be able to supply their own payment hash? cc @jkczyz

Reply:
Yeah, I think it could be moved there, though I'm not sure offhand if that would require changes to `OffersMessageFlow` to allow using `create_inbound_payment`.

For Fedimint (or more generally), we'd likely introduce an event model for `OfferMessageFlow` to allow them to supply a payment hash. @shaavan and I chatted about this, but details are still in the early stage.

Reply:
Makes sense! Addressed in [pr3639.13](https://github.com/shaavan/rust-lightning/commits/pr3639.13)
Thanks!